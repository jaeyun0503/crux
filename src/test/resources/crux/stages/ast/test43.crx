// test functions with value must have return

fn validReturn1(a: int) -> int {
  if (true) {
    if (false) {
      a = 3;
    } else {
      while (false) {
        return a;
      }
    }
    return 4;
  } else {
    return a;
  }
}

fn main() {
  printInt(3);
  printInt(validReturn1(6));
}
