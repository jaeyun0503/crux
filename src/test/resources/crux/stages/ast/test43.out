
validReturn1(a) {
  if (true) {
    if (false) {
      a = 3;
    } else {
      while (false) {
        return a;
      }
    }
    return 4;
  } else {
    return a;
  }
}
main() {
  printInt(3);
  printInt(validReturn1(6));
}

